### PR TITLE
docs:fixing broken links in readme

### LINKED ISSUE #10983: Doc bug: 404 for "Emissary Executor" in https://github.com/kubeflow/pipelines/blob/master/README.md
### Environment

*  How do you deploy Kubeflow Pipelines (KFP)?

*  KFP version:

*  KFP SDK version:

### Labels

---

Impacted by this bug? Give it a 👍.

### PR DESCRIPTION
**Pull request by Fiona-Waters**
**Description of your changes:**
Fixes #10983 and other broken links in the repository readme.

I have updated some broken links in the readme. I also removed the section referring to the versioning policy and feature stages as those links do not exist in this repo - please let me know if I can link to them somewhere else.

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,22 +17,20 @@ The Kubeflow pipelines service has the following goals:
 
 ## Installation
 
-* Install Kubeflow Pipelines from choices described in [Installation Options for Kubeflow Pipelines](https://www.kubeflow.org/docs/pipelines/installation/overview/).
+* Install Kubeflow Pipelines from choices described in [V1-Installation Options for Kubeflow Pipelines](https://www.kubeflow.org/docs/components/pipelines/legacy-v1/installation/overview/) and [V2-Operator Guides Installation](https://www.kubeflow.org/docs/components/pipelines/operator-guides/installation/).
```
Comment (hbelmiro) [inline README.md:20]: I wonder if we shouldn't use https://www.kubeflow.org/docs/components/pipelines/operator-guides/installation/ instead. I understand though it lacks a lot of information and the info in the v1 docs helps a lot.
I'd like to know the opinion of others.

cc @HumairAK @rimolive @gregsheremeta

File: README.md [hunk 2]
```diff
@@ -17,22 +17,20 @@ The Kubeflow pipelines service has the following goals:
...
+* The Docker container runtime has been deprecated on Kubernetes 1.20+. Kubeflow Pipelines has switched to use [Emissary Executor](https://www.kubeflow.org/docs/components/pipelines/legacy-v1/installation/choose-executor/#emissary-executor) by default from Kubeflow Pipelines 1.8. Emissary executor is Container runtime agnostic, meaning you are able to run Kubeflow Pipelines on Kubernetes cluster with any [Container runtimes](https://kubernetes.io/docs/setup/production-environment/container-runtimes/). 
 
 ## Documentation
 
-Get started with your first pipeline and read further information in the [Kubeflow Pipelines overview](https://www.kubeflow.org/docs/components/pipelines/introduction/).
+Get started with your first pipeline and read further information in the [Kubeflow Pipelines overview](https://www.kubeflow.org/docs/components/pipelines/overview/).
 
-See the various ways you can [use the Kubeflow Pipelines SDK](https://www.kubeflow.org/docs/pipelines/sdk/sdk-overview/).
+See the various ways you can [use the Kubeflow Pipelines SDK](https://www.kubeflow.org/docs/components/pipelines/legacy-v1/sdk/sdk-overview/).
```
Comment (hbelmiro) [inline README.md:28]: We have a specific documentation for the SDK now.
```suggestion
See the various ways you can [use the Kubeflow Pipelines SDK](https://kubeflow-pipelines.readthedocs.io/en/stable/).
```

File: README.md (hunk 1 above)
Comment (gregsheremeta) [reply README.md:20]: I think this section should call out that you can install Kubeflow as a whole, or you can install Kubeflow (v2) as a standalone. So yeah, I think @hbelmiro 's suggested [link](https://www.kubeflow.org/docs/components/pipelines/operator-guides/installation/) makes the most sense (although I'd copy a bit of the blurb from that page too).

I'm generally not a fan of still linking to old v1 stuff from v2 docs. v1 should be relegated to its hidden corner of the docs.

**Comment by gregsheremeta** [comment]
Thanks @Fiona-Waters !

File: README.md (hunk 1 above)
Comment (Fiona-Waters) [reply README.md]: Thanks @hbelmiro @gregsheremeta!
I have updated this is a new commit which I can squash when the PR is ready.

File: README.md (hunk 2 above)
Comment (Fiona-Waters) [reply README.md]: Updated, thank you.

**Review by hbelmiro — COMMENTED** [review]
Thank you @Fiona-Waters.

/lgtm

**Comment by hbelmiro** [comment]
@chensun can you PTAL?

**Review by hbelmiro — COMMENTED** [review]
/lgtm

File: README.md [hunk 3]
```diff
@@ -17,22 +17,21 @@ The Kubeflow pipelines service has the following goals:
 
 ## Installation
 
-* Install Kubeflow Pipelines from choices described in [Installation Options for Kubeflow Pipelines](https://www.kubeflow.org/docs/pipelines/installation/overview/).
+* Kubeflow Pipelines can be installed as part of the [Kubeflow Platform](https://www.kubeflow.org/docs/started/installing-kubeflow/#kubeflow-platform). Alternatively you can deploy [Kubeflow Pipelines (V2)](https://www.kubeflow.org/docs/components/pipelines/operator-guides/installation/) as a standalone service.
```
Comment (chensun) [inline README.md:20]: nit: I'd suggest remove 'V2' here, the standalone deployment instruction applies to all versions.

**Review by chensun — COMMENTED** [review]
One nitpick, otherwise LGTM

File: README.md (hunk 3 above)
Comment (Fiona-Waters) [reply README.md]: Updated and commits squashed. Thanks @chensun

**Review by chensun — APPROVED** [review]
/lgtm
/approve

**Pull request merged**